python/lsst/daf/butler/core/registry.py

```diff
@@ -76,6 +78,185 @@ class RegistryConfig(ConfigSubset):
+    I)   RegistryConfig for explicitly defined keys,
+    II)  'db' key in RegistryConfig and
+    III) and ~/.lsst/db-auth.paf file.
+
+    When not stated explicitly, or included in the 'db' string, dialect and
+    driver are inferred from targeted registry class defaults. The returned
+    connection string has the following format:
+
+        `dialect+driver://(username:password@?)(host?)/(dbname?)(:port?))`
+
+    Where host, port, username and password are not returned for sqlite
+    dialects, making dbname effectively the path to the DB. Username and
+    password are optional in other cases, such as when credentials hadling
+    are be left to the DB itself.
+    """
+    requiredKeys = ('dialect', 'driver', 'username', 'password', 'host', 'port',
+                    'dbname')
+    """Keys required to construct a fully qualified connection string."""
+
+    @classmethod
+    def _dictFromRegistryConf(cls, regConf):
+        """Looks for explicit (key, value) pairs required for connection string
+        in a Yaml config. Missing values are returned as None.
+
+        Explicitly given keywords take precedence over values found in 'db'
+        string.
+
+        Parameters
+        ----------
+        regConf : `lsst.daf.butler.Config`, `str`
+            Config or path to Yaml config file to parse.
+
+        Returns
+        -------
+        dsnKwargs : `dict`
+            Dictionary containing found values (or None) for keys required for
+            a connection string.
+        """
+        #
+        #    Parse the explicit keywords in the config file.
+        #
+        #
+        regConf = RegistryConfig(regConf)
+        explicitKwargs, conStrKwargs = dict(), dict()
+        for key in cls.requiredKeys:
+            explicitKwargs[key] = regConf.get(key)
+            conStrKwargs[key] = None
+
+        # ensure host is only the hostname and not mistaken for the 'db' key as
+        # there is no additional parsing for host unlike 'db'.
+        if explicitKwargs['host'] is not None:
+            parsed = urlparse.urlparse(explicitKwargs['host'])
+            if parsed.scheme or parsed.netloc or ':' in parsed.path:
+                raise ValueError(("Invalid 'host' key, 'host' should only be "
+                                  "the hostname. For URLs/URIs use 'db' key. "
+                                  f"Received host: {explicitKwargs['host']}. "))
+
+        #
+        #    Parse the keywords from the connection string.
+        #
+        parsed = urlparse.urlparse(regConf['db'])
+        registryCls = doImport(regConf['cls'])
+
+        # URLs with scheme (f.e. dialect(+driver?)://(netloc?)/(path?)(:port?))
+        if parsed.scheme and not parsed.netloc:
+            # driver is infered from target registry class when not explicit
+            if '+' in parsed.scheme:
+                conStrKwargs['dialect'], conStrKwargs['driver'] = parsed.scheme.split('+')
```

Comment:
This will crash if there is more than one `+` in scheme. I'd suggest `partition()` instead of `split()`.